kratos/geometries/triangle_3d_3.h

```diff
@@ -974,6 +974,123 @@ template<class TPointType> class Triangle3D3
+    *         keep more possibilities within the interface.
+    *         0 -> failed
+    *         1 -> converged
+    */
+    KRATOS_DEPRECATED_MESSAGE("This method is deprecated. Use either \'ProjectionPointLocalToLocalSpace\' or \'ProjectionPointGlobalToLocalSpace\' instead.")
+    int ProjectionPoint(
+        const CoordinatesArrayType& rPointGlobalCoordinates,
+        CoordinatesArrayType& rProjectedPointGlobalCoordinates,
+        CoordinatesArrayType& rProjectedPointLocalCoordinates,
+        const double Tolerance = std::numeric_limits<double>::epsilon()
+        ) const override
+    {
+        KRATOS_WARNING("ProjectionPoint") << "This method is deprecated. Use either \'ProjectionPointLocalToLocalSpace\' or \'ProjectionPointGlobalToLocalSpace\' instead." << std::endl;
+
+        ProjectionPointGlobalToLocalSpace(rPointGlobalCoordinates, rProjectedPointLocalCoordinates, Tolerance);
+
+        this->GlobalCoordinates(rProjectedPointGlobalCoordinates, rProjectedPointLocalCoordinates);
+
+        return 1;
+    }
+
+    /**
+     * @brief Projects a point onto the geometry
+     * Projects a certain point on the geometry, or finds the closest point, depending on the provided initial guess.
+     * The external point does not necessary lay on the geometry.
+     * It shall deal as the interface to the mathematical projection function e.g. the Newton-Raphson.
+     * Thus, the breaking criteria does not necessarily mean that it found a point on the surface, if it is really
+     * the closest if or not.
+     * It shows only if the breaking criteria, defined by the tolerance is reached.
+     * This function requires an initial guess, provided by rProjectionPointLocalCoordinates.
+     * This function can be a very costly operation.
+     * @param rPointLocalCoordinates Local coordinates of the point to be projected
+     * @param rProjectionPointLocalCoordinates Projection point local coordinates. This should be initialized with the initial guess
+     * @param Tolerance Accepted orthogonal error
+     * @return int 0 -> failed
+     *             1 -> converged
+     */
+    int ProjectionPointLocalToLocalSpace(
+        const CoordinatesArrayType& rPointLocalCoordinates,
+        CoordinatesArrayType& rProjectionPointLocalCoordinates,
+        const double Tolerance = std::numeric_limits<double>::epsilon()
+        ) const override
+    {
+        double sum_coordinates = 0.0;
+        for(unsigned int i = 0 ; i < 2 ; i++){ // It's a triangle, so only (local) xi and eta
+            if (rProjectionPointLocalCoordinates[i] < 0.0) { // Clipping to zero
+                rProjectionPointLocalCoordinates[i] = 0.0;
+            }
+            sum_coordinates += rProjectionPointLocalCoordinates[i];
+        }
+        
+        // Clipping to line y=1-x
+        if (sum_coordinates>1.0){
+            for(unsigned int i = 0 ; i < 2 ; i++){ 
+                rProjectionPointLocalCoordinates[i] /= sum_coordinates;

```

Comment:
Maybe I'm missing something, but aren't you assuming in here that `rProjectionPointLocalCoordinates` is initialized before? In other words, what would happen if you dont reach the statement in current line 1048?

Reply:
This is one year old, I don't even remember what this code does

Reply:
Copy was missing

Reply:
> Copy was missing

What does this mean?

Reply:
The formula was okay, but I needed to copy values first, see https://github.com/KratosMultiphysics/Kratos/pull/11051/commits/0915098c6084d142c887eb170b595edb710fb47c